cogs/make_applicant.py

```diff
@@ -51,15 +51,20 @@ async def _perform_make_applicant(self, ctx: TeXBotApplicationContext, applicant
 
         AUDIT_MESSAGE: Final[str] = f"{ctx.user} used TeX Bot Command \"Make User Applicant\""
 
-        await applicant_member.add_roles(applicant_role, reason=AUDIT_MESSAGE)
-
-        logger.debug("Applicant role given to user %s", applicant_member)
-
         if guest_role in applicant_member.roles:
             await applicant_member.remove_roles(guest_role, reason=AUDIT_MESSAGE)
             logger.debug("Removed Guest role from user %s", applicant_member)
 
 
+        if applicant_role not in applicant_member.roles:
+            await applicant_member.add_roles(applicant_role, reason=AUDIT_MESSAGE)
+            logger.debug("Applicant role given to user %s", applicant_member)
+            await initial_response.edit(
+                content=(
+                    ":hourglass: User already has the applicant role!"
+                ),
+            )
```

Comment:
2 issues with this message.

1. The edit command is called but will quickly be replaced by the completed message (line 92). It is bad UX to hope that the user sees this message pop up momentarily before being replaced again
2. The message does not match the logic of the `if` statement above. This message about the user already having the applicant role would be sent only if the user doesn't already have the applicant role.